chains/celo/listener.go

```diff
@@ -33,3 +40,21 @@ func (l *listener) start() error {
 func (l *listener) close() {
 	l.conn.Close()
 }
+
+func (l *listener) getTransactionBlockHash(hash common.Hash) (tx *types.Transaction) {
+	client := l.conn.Client()
+	blockHash, _, err := client.TransactionByHash(context.Background(), hash)
+	if err != nil {
+		fmt.Errorf("unable to get BlockHash: %s", err)
+	}
+	return blockHash
+}
+
+func (l *listener) getBlockTransactionsByHash(hash common.Hash) *types.Block {
```

Comment:
same here, we're returning the entire block body rather than the transaction field from the body, making this somewhat unclear.